sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -43,6 +43,45 @@ class CastSuite extends SparkFunSuite with ExpressionEvalHelper {
     checkEvaluation(cast(v, Literal(expected).dataType), expected)
   }
 
+  private def checkNullCast(from: DataType, to: DataType): Unit = {
+    checkEvaluation(Cast(Literal.create(null, from), to), null)
+  }
+
+  test("null cast") {
+    val numericTypes = Seq(IntegerType, ShortType, ByteType, LongType,
```

Comment:
I found myself writing these lists a lot in my own tests, so I defined some similar ones as constants in `DataTypeTestUtils`.